railties/lib/rails/application/finisher.rb

```diff
@@ -93,6 +94,28 @@ module Finisher
         end
       end
 
+      initializer :notify_early_loaded_component do
+        next if config.eager_load
+
+        rails_components = [
+          :action_view,
+          :action_cable,
+          :action_controller,
+          :active_job,
+          :action_mailer,
+          :active_record
+        ]
+        loaded_components = ActiveSupport.loaded_components.keys & rails_components
+        if loaded_components.any?
+          $stderr.puts(<<~EOM)
+            Your application (or a dependencies) is making use of Rails component too early on boot.
+            This can cause various issues (such as ignoring some configuration values you set) and increase boot time.
+            To discard this warning, use lazy load hooks instead.
```

Comment:
Maybe point to the documentation?

Reply:
Point to https://guides.rubyonrails.org/engines.html#active-support-on-load-hooks

Reply:
In the Edge Guides it's: https://edgeguides.rubyonrails.org/engines.html#load-and-configuration-hooks